storage/remote/write_handler_test.go

```diff
@@ -165,6 +167,65 @@ func TestCommitErr(t *testing.T) {
 	require.Equal(t, "commit error\n", string(body))
 }
 
+func BenchmarkRemoteWriteOOOSamples(b *testing.B) {
+	dir := b.TempDir()
+	opts := tsdb.DefaultHeadOptions()
+	opts.ChunkDirRoot = dir
+	opts.OutOfOrderCapMax.Store(30)
+	opts.OutOfOrderTimeWindow.Store(120 * time.Minute.Milliseconds())
+
+	h, err := tsdb.NewHead(nil, nil, nil, nil, opts, nil)
+	require.NoError(b, err)
+	b.Cleanup(func() {
+		require.NoError(b, h.Close())
+	})
+	require.NoError(b, h.Init(0))
```

Comment:
Last comment before we can merge. Outside of tsdb package, we should always use tsdb.DB for testing, because that is what is used as a storage backend, and not just the Head. So tsdb.Open instead of tsdb.NewHead.

Reply:
Addressed here https://github.com/prometheus/prometheus/pull/11615/commits/8acfca91ff59196dfaa8e5ad663aa11e8e04ef52